### ISSUE TITLE
{{ nocache }} Tag inside a antlers component 

### ISSUE THREAD

**Issue by exophunk** [Stillat/antlers-components#5]
## The Problem
Using `{{ nocache }}` anywhere inside a `<a-partial-name></a-partial-name>` component causes this Error:

`Serialization of 'Closure' is not allowed` in vendor / laravel / framework / src / Illuminate / Cache / FileStore.php : 84
Happens as soon as the page is rendered. Using `{{ partial:name }}` does **not** cause the bug.

## How to reproduce
1. Use `{{ nocache }}` inside a partial included via `<a-partialname>`
2. Enable full static caching
3. Load the page
4. Compare with `{{ partial:name }}` where it doesn't break

## Context
We are using "Antlers components" all over our website. In fact I've replaced every `{{ partial:name }}` tag with `<a-partial>`. 
The main reason is because your components get variable scoping right, where as the normal `{{ partial }}` tags have super weird inconsistent hierarchy of variables leaking from "child" partials to everywhere else and that confused me. We're using it for livewire components too (all antlers)

We're using the `<a-partial>` tag in 366 places and it works absolutely great. We're also using full static caching successfully. I love it.

Now, we've added User Login functionality, which requires us to use `{{ nocache }}` in certain places to not show cached "logged in state" to not authenticated users.

I love your components and I'm also afraid switching back to `{{ partial }}` as it doesn't really work properly in my eyes. But not being able to use nocache makes it very difficult. And also we can't afford to disable static caching.

I've no idea if this is an easy problem to solve or something not possible?

**Comment by JohnathonKoster** [comment]
Do you have a very simple sample repository that replicates this that I could look at? Its possible that it has to do with the component attributes/etc. that is happening under the hood. The only fix to this would be to mitigate it, as serialization of Closures is not possible.

**Comment by exophunk** [comment]
Sorry for the late response. Just tried to set up a clean repository with a easy to understand case. While doing this, I saw that using {{ nocache }} inside <a-partial> alone isn't a problem and works fine.

I wasn't able yet to pin down what exactly was the issue in our case. Must be because of a specific data structure inside the nocache tag. But we mitigated otherwise and your plugin doesn't seem affected for simple cases. So I'd say we can close this for now.
